### PR TITLE
Remove one level of indirection

### PR DESCRIPTION
**Pull request by omni-viral**

### CODE REVIEW

File: amethyst_assets/src/loader.rs [hunk 1]
```diff
@@ -108,9 +108,9 @@ unsafe impl<A, E> Sync for AssetFutureCell<A, E> {}
 /// `ThreadPool`.
 pub struct Loader {
     contexts: FnvHashMap<TypeId, Box<Any>>,
-    directory: Arc<Box<AnyStore>>,
+    directory: Arc<AnyStore>,
```
Comment (torkleyy) [inline amethyst_assets/src/loader.rs]: Oh... I didn't have in mind that `Arc` allocates.

File: amethyst_assets/src/store/mod.rs [hunk 2]
```diff
@@ -50,24 +50,26 @@ impl<T> AnyStore for T
     }
 }
 
-impl<T> Store for Box<T>
-    where T: AnyStore + ?Sized
+impl<T, S> Store for T
+    where T: ::std::ops::Deref<Target=S>,
```
Comment (torkleyy) [inline amethyst_assets/src/store/mod.rs]: Could you please add an import?

File: amethyst_assets/src/store/mod.rs [hunk 3]
```diff
@@ -50,24 +50,26 @@ impl<T> AnyStore for T
...
+          S: AnyStore + ?Sized
 {
     type Error = BoxedErr;
 
     fn modified(&self, category: &str, id: &str, ext: &str) -> Result<u64, Self::Error> {
-        T::modified(self, category, id, ext)
+        S::modified(self.deref(), category, id, ext)
```
Comment (brendanzab) [inline amethyst_assets/src/store/mod.rs]: Why not use `*self` here instead of calling `deref` directly?

Comment (omni-viral) [inline amethyst_assets/src/store/mod.rs]: I have no preferences here. I change if people want.
`self` would be enough thanks to deref coercion.

File: amethyst_assets/src/loader.rs (hunk 1 above)
Comment (omni-viral) [inline amethyst_assets/src/loader.rs]: It even doesn't matter if it allocates. What does is implementation of `CoerceUsized`.

**Pull request merged**